moveit_core/planning_interface/src/planning_interface.cpp

```diff
@@ -117,6 +117,18 @@ void PlannerManager::getPlanningAlgorithms(std::vector<std::string>& algs) const
 void PlannerManager::setPlannerConfigurations(const PlannerConfigurationMap& pcs)
 {
   config_settings_ = pcs;
+
+  // Construct default configurations for planning groups that don't have configs already passed in. This is necessary
+  // since moveit_cpp can only use planners with a config for a given joint group
+  for (const moveit::core::JointModelGroup* group : robot_model_->getJointModelGroups())
+  {
+    if (config_settings_.find(group->getName()) == config_settings_.end())
+    {
+      planning_interface::PlannerConfigurationSettings empty_config;
+      empty_config.name = empty_config.group = group->getName();
+      config_settings_[empty.name] = empty;
+    }
+  }
```

Comment:
Sorry, you reviewed to fast, that change did not work. The problem is that initialize() function can easily be overwritten and thus compatibility with moveit_cpp be destroyed. The way I implemented it would have added a default empty config for all planning plugins with `setPlannerConfigurations`. But that did not work because the robot model is not available in this function. 

I think the planner plugins should be modified because the flaw right now is that pconfig is not used correctly (as expected by moveit_cpp) by the plugins. If we hide this implementation detail away in the base class, I fear future plugin implementations will repeat the same error,

Reply:
> pconfig is not used correctly (as expected by moveit_cpp) by the plugins.

Can you please detail this? As far as I can see, moveit_cpp just generates a "shortcut" map from these configs.
If there are no config settings, why should a plugin declare them as empty? I think we should first agree on the semantics, before implementing anything further...

Reply:
My preference would be to use the current solution and open an issue to re-evaluate the PlannerConfigurationSettings, since it is only properly used by OMPL at the moment. Maybe it would be better to remove/refactor it and the pipeline detection in moveit_cpp but I think this is out of the scope of this PR.

Reply:
https://github.com/ros-planning/moveit2/pull/1522